repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeFilter.scala

```diff
@@ -19,7 +19,200 @@
+
+        MDC.put(TracingKey.TRACING_KEY, traceGUID)
+
+        val doFilterChain = TryWith(filterContextList)(_ => {
+          // Ensure the request URI is a valid URI
+          // This object is only being created to ensure its validity.
+          // So it is safe to suppress warning squid:S1848
+          new URI(wrappedRequest.getRequestURI)
+
+          if (tracingHeaderConfig.exists(_.isEnabled)) {
+            if (StringUtils.isBlank(wrappedRequest.getHeader(TRACE_GUID))) {
+              wrappedRequest.addHeader(TRACE_GUID, TracingHeaderHelper.createTracingHeader(traceGUID, wrappedRequest.getHeader(VIA)))
+            }
+            if (tracingHeaderConfig.exists(_.isSecondaryPlainText)) {
+              TraceIdLogger.trace("Adding plain text trans id to request: {}", traceGUID)
+              wrappedRequest.replaceHeader(REQUEST_ID, traceGUID)
+            }
+            val tracingHeader = wrappedRequest.getHeader(TRACE_GUID)
+            TraceIdLogger.info("Tracing header: {}", TracingHeaderHelper.decode(tracingHeader))
+            wrappedResponse.addHeader(TRACE_GUID, tracingHeader)
+          }
+
+          new ReposeFilterChain(
+            filterContextList.filterContexts,
+            chain,
+            filterContextList.bypassUriRegex,
+            optMetricRegistry,
+            tracer
+          ).doFilter(wrappedRequest, wrappedResponse)
+        })
+
+        doFilterChain match {
+          case Success(_) =>
+            logger.trace("{} -- Successfully processed request", nodeId)
+          case Failure(use: URISyntaxException) =>
+            logger.debug(s"$nodeId -- Invalid URI requested: ${wrappedRequest.getRequestURI}", use)
+            wrappedResponse.sendError(SC_BAD_REQUEST, "Error processing request", true)
+          case Failure(e: Exception) =>
+            logger.error(s"$nodeId -- Issue encountered while processing filter chain.", e)
+            wrappedResponse.sendError(SC_BAD_GATEWAY, "Error processing request", true)
+          case Failure(t: Throwable) =>
+            logger.error(s"$nodeId -- Error encountered while processing filter chain.", t)
+            wrappedResponse.sendError(SC_BAD_GATEWAY, "Error processing request", true)
+            setViaClearCommitAndClose(wrappedRequest, wrappedResponse, scope, startTime)
+            throw t
+        }
+        setViaClearCommitAndClose(wrappedRequest, wrappedResponse, scope, startTime)
+        logger.trace("ReposeFilter returning response...")
+      case None =>
+        logger.error("ReposeFilter has not yet initialized...")
+        response.asInstanceOf[HttpServletResponse].sendError(SC_INTERNAL_SERVER_ERROR, "ReposeFilter not initialized")
+    }
+  }
+
+  private def setViaClearCommitAndClose(wrappedRequest: HttpServletRequestWrapper, wrappedResponse: HttpServletResponseWrapper, scope: Scope, startTime: Long): Unit = {
+    if (!wrappedResponse.isCommitted) {
+      responseHeaderService.setVia(wrappedRequest, wrappedResponse)
```

Comment:
I think the right choice is to uncommit it in this block, and then outside of the block always write the via header.

Reply:
Agreed.

Reply:
Updated.